src/main/java/core/basesyntax/dao/impl/FruitStorageDaoImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.dao.impl;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.db.FruitStorage;
+import java.util.Map;
+import core.basesyntax.model.FruitTransaction;
+
+public class FruitStorageDaoImpl implements FruitStorageDao {
+    @Override
+    public void addTransaction(FruitTransaction transaction) {
```

Comment:
let's accept fruit name and quantity, not whole transaction

Reply:
I can do this. But there is only one reason why I create FruitTransaction it is to pass it in FruitStorageDaoImpl to fill in the storage. If I pass fruit name and quantity it this method I have to remove package "model" at all and FruitTransactionService  with its implementation as well. If I'm not mistaken, sample of FruitTransaction was done in file "Try to avoid common mistake". In my first version of the task I was storing FruitTransaction object in List storage. After rejection I change it with Map with name of fruits as keys and amount as values but continue using FruitTransaction object to fill in the storage. If I change this method why do I need FruitTransaction?

Reply:
you need `FruitTransaction` to conveniently "pack" parsed data into some entity and process it later
the whole process should look like this:
reading data (`ReaderService`) 
-> parsing data (here combine `FruitTransactionService` and `ParsingService` into one, that will accept each `String` from the data list and create `FruitTransaction` based on it) 
-> processing transactions (create some service that will execute transactions)
```
    public void executeTransactions(List<FruitTransaction> transactions) {
        for (FruitTransaction transaction : transactions) {
            OperationHandler handler = operationStrategy.get(transaction.getOperation());
            handler.execute(transaction);
        }
    }
```
something like this
handlers in example are yours `AmountService`'s, they should call dao methods and pass data into it
for example you can have method in dao that will add quantity to specified fruit, so it handlers you can pass 100 if you want to add 100 units or -100 if you want to subtract 100 units
-> then creating a report based on what data is in storage (`ReportService`)
-> writing it to file (`WriterService`)

Hope you get the idea! If you still have some questions ask them in slack private channel, that way I will see them much faster)
Or come to QnA